src/System.Diagnostics.Process/src/System/Diagnostics/Process.FreeBSD.cs

```diff
@@ -64,8 +64,16 @@ public TimeSpan PrivilegedProcessorTime
         }
 
         /// <summary>Gets parent process ID</summary>
-        private int ParentProcessId =>
-            throw new PlatformNotSupportedException();
+        private unsafe int ParentProcessId
+        {
+            get
+            {
+                EnsureState(State.HaveNonExitedId);
+
+                Interop.Process.kinfo_proc* stat = Interop.Process.GetProcInfo(_processId, false, out _);
```

Comment:
This is a memory leak. It needs to be freed.

Reply:
Added a try-finally with free call.